cedars/app/auth.py

```diff
@@ -68,26 +77,40 @@ def __init__(self, data):
     def get_id(self):
         return str(self.username)
 
-
 @bp.route("/register", methods=["GET", "POST"])
 def register():
     """Register a new user"""
+
     if request.method == 'POST':
         username = request.form.get("username")
         password = request.form.get("password")
         confirm_password = request.form.get("confirm_password")
         is_admin = request.form.get("isadmin") == "on"
-        print(f"Registering user {username} as admin: {is_admin}")
+        logger.info(f"Registering user {username} as admin: {is_admin}")
+
+        password_policy = PasswordPolicy(
+            min_length=8,
+            min_uppercase=2,
+            min_lowercase=2,
+            min_digits=2,
+            min_special=2,
+            special_chars="!@#$%^&*[]{}()~`,./<>?;:'\"-_+",
+            allow_spaces=False
+        )
+        password_result, password_issues = password_policy.check_password(password)
 
         error = None
         existing_user = db.get_user(username)
         if password != confirm_password:
             error = 'Passwords do not match.'
         elif not username or not password or len(username.strip()) == 0 or len(password.strip()) == 0:
             error = "Username and password are required."
-
         elif existing_user or username.lower() in ['cedars', 'pines']:
             error = 'Username already exists or reserved. Please choose a different one.'
+        elif password_result is False:
+            # If the password is not in compliance with our policy,
+            # display the first issue found with this password
+            error = password_issues[0]
```

Comment:
you can display all issues at once by doing something like `"\n".join(password_issues)`